### ISSUE TITLE
Check certificate error : :Invalid certificate chain with certificate in signature.:<-3>

### ISSUE THREAD

**Issue by BlytheScythe** [reisxd/TizenBrew#17]
Greetings,

I've tried updating to the latest version of the TizenBrew on my Samsung TV via latest updater, but I've run into the following error (the same one mentioned in the title:

https://i.imgur.com/OQzAxLr.jpeg

Here's the log from the updater app as well:

> Service started.
> Please open up http://localhost:8083 in your browser.
> (node:13140) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
> (Use `tizenbrew-updater-win-x64 --trace-deprecation ...` to show where the warning was created)
> ADB connection established

I've tried to upload and install the latest version via Tizen Studio, but I get the same error. Here's the log output:

> C:\tizen-studio\tools\ide\bin>tizen install -n TizenBrewStandalone-New.wgt
> Transferring the package...
> Transferred the package: C:\tizen-studio\tools\ide\bin\TizenBrewStandalone-New.wgt -> /home/owner/share/tmp/sdk_tools/tmp
> Installing the package...
> --------------------
> Platform log view
> --------------------
> install xvvl3S1bvH.TizenBrewStandalone
> package_path /home/owner/share/tmp/sdk_tools/tmp/TizenBrewStandalone-New.wgt
> app_id[xvvl3S1bvH.TizenBrewStandalone] install start
> app_id[xvvl3S1bvH.TizenBrewStandalone] installing[2]
> app_id[xvvl3S1bvH.TizenBrewStandalone] installing[4]
> app_id[xvvl3S1bvH.TizenBrewStandalone] installing[6]
> app_id[xvvl3S1bvH.TizenBrewStandalone] installing[8]
> app_id[xvvl3S1bvH.TizenBrewStandalone] installing[10]
> app_id[xvvl3S1bvH.TizenBrewStandalone] installing[12]
> app_id[xvvl3S1bvH.TizenBrewStandalone] installing[14]
> app_id[xvvl3S1bvH.TizenBrewStandalone] installing[16]
> app_id[xvvl3S1bvH.TizenBrewStandalone] installing[18]
> app_id[xvvl3S1bvH.TizenBrewStandalone] installing[20]
> app_id[xvvl3S1bvH.TizenBrewStandalone] installing[22]
> app_id[xvvl3S1bvH.TizenBrewStandalone] install failed[118, -12], reason: Check certificate error : :Invalid certificate chain with certificate in signature.:<-3>
> spend time for wascmd is [809]ms
> Failed to install Tizen application.
> Total time: 00:00:08.537

I've tried using the old version, but I've got another error:

> C:\tizen-studio\tools\ide\bin>tizen install -n TizenBrewStandalone-Old.wgt
> Transferring the package...
> Transferred the package: C:\tizen-studio\tools\ide\bin\TizenBrewStandalone-Old.wgt -> /home/owner/share/tmp/sdk_tools/tmp
> Installing the package...
> --------------------
> Platform log view
> --------------------
> install xvvl3S1bvH.TizenBrewStandalone
> package_path /home/owner/share/tmp/sdk_tools/tmp/TizenBrewStandalone-Old.wgt
> app_id[xvvl3S1bvH.TizenBrewStandalone] install start
> app_id[xvvl3S1bvH.TizenBrewStandalone] installing[2]
> app_id[xvvl3S1bvH.TizenBrewStandalone] installing[4]
> app_id[xvvl3S1bvH.TizenBrewStandalone] installing[6]
> app_id[xvvl3S1bvH.TizenBrewStandalone] installing[8]
> app_id[xvvl3S1bvH.TizenBrewStandalone] installing[10]
> app_id[xvvl3S1bvH.TizenBrewStandalone] installing[12]
> app_id[xvvl3S1bvH.TizenBrewStandalone] installing[14]
> app_id[xvvl3S1bvH.TizenBrewStandalone] installing[16]
> app_id[xvvl3S1bvH.TizenBrewStandalone] installing[18]
> app_id[xvvl3S1bvH.TizenBrewStandalone] installing[20]
> app_id[xvvl3S1bvH.TizenBrewStandalone] installing[22]
> app_id[xvvl3S1bvH.TizenBrewStandalone] installing[25]
> app_id[xvvl3S1bvH.TizenBrewStandalone] install failed[118, -11], reason: Author certificate not match :
> spend time for wascmd is [826]ms
> Failed to install Tizen application.
> Total time: 00:00:05.300

I'm not sure what to do to fix that error as I'm not familiar with how Tizen Studio really works, I've just followed the instructions mentioned [here](https://github.com/reisxd/TizenBrew/blob/main/docs/README.md).

Previously I have installed TizenBrew on this TV of mine, but for some reason I got the following message whenever I opened up TizenBrew:

> Error: Could not connect to the server. Are you sure you changed the Host PC IP to 127.0.0.1? If you have, hold the power button till you see the Samsung logo.

I'm pretty sure I've went into Apps and pressed 1-2-3-4-5 and set the IP address there, pressed OK and powered off the TV by holding the power button until I saw the Samsung logo (well, QLED logo). Regardless of what I do, I still get the same error.

TizenTube works, only sometimes it doesn't autoload. But it works, nevertheless.

My TV model number is: QE55Q67CAUXXH.
Software version: T-NKLCDEUC-1310.3, E2371410, BT-S.

Let me know what I can do to fix this or if you need any more info.
Thanks in advance!

**Comment by reisxd** [comment]
Your TV is not Tizen 8. You should install the "Old" package.

**Comment by BlytheScythe** [comment]
Thanks for the quick reply and instructions!

I've managed to uninstall the app first and successfully installed the updated one via updater.

Unfortunately, now the TizenTube isn't working anymore. I can add the tizentube module, but if I set it to autolaunch, the tizenbrew app keeps opening and closing. Once, I saw these errors: Invalid JSON and failed to connect to debugger, before the app closed itself again. The other time I saw that no module has been loaded, but when the module is displayed, I get the pop-up saying: You can't launch modules while the service hasn't connected yet. Weirdest of all, on the upper right corner it says Service is already running.

Is it possible to do a clean install? I wonder if there are some leftovers from the previous installs, maybe due to all my failed installs?

**Comment by BlytheScythe** [comment]
This morning, after I've reset my TV to factory defaults, I've tried to reinstall TizenBrew app via Updater and for some reason it wouldn't connect to the TV for some reason. I've double-checked the IP address and that I've enabled dev mode with the correct host PC IP, but to no avail.

I've tried installing the app via TizenStudio as I did so in the past and I was successful this time around.

Unfortunately, I'm still unable to use TIzenTube as I'm getting "Invalid JSON" error, maybe due to the leftovers from previous attempts? I believe that they aren't cleared after factory reset? What can I do about it?

Also, should I open another issue seeing that this one's closed?
